### DISCUSSION TITLE
MuJoCo model for the JVRC1 (virtual) humanoid robot

### DISCUSSION THREAD

**Discussion by rohanpsingh** [google-deepmind/mujoco#684] [Show and tell]
## We introduce :robot: [jvrc_mj_description](https://github.com/isri-aist/jvrc_mj_description) :robot: 

https://user-images.githubusercontent.com/16384313/210514033-4ae13232-acf9-4534-8385-2612cd6c0e3a.mp4

https://user-images.githubusercontent.com/16384313/210514554-d7606bcc-09be-4b9f-8e5a-c96c1737c77d.mp4

**What is this?**
JVRC1 is a virtual humanoid robot created for the Japan Virtual Robotics Challenge (held in 2015). The original VRML model files exist [here](https://github.com/jvrc/model). We have ported the model to MuJoCo along with the textures and minor modifications to the grippers: [jvrc_mj_description](https://github.com/isri-aist/jvrc_mj_description).

The robot has a total of 44 (34 actuated + 10 passive) joints making it a 50-DoF robot (`nq=51`, `nv=50`, `nu=34`). Each gripper has 1 actuated joint and 5 "mimic" joints. Each end-effector (feet and hands) has a 6-DoF force-torque sensor. There is one IMU sensor placed at the root body.

**Why is this important? This is just a virtual robot?**
While there exist several other models such as the mujoco Humanoid and the Humanoid_CMU widely used in the research community, we need a robot model that is more closely modeled after a _humanoid robot_ than an actual human. This is important for creating and evaluating controllers meant to be deployed on real humanoid robots. In my opinion, it is rare to find open-source model files for physical humanoid robots of this size.

Yes, JVRC1 is a virtual robot with no corresponding physical form but it is close to one of the real humanoid robots of the "HRP" series, that some of you may already be familiar with, in terms of - kinematic structure, link masses and inertias, armature values, joint ranges, etc. We can even introduce actuator torque limits in the future.

I hope this can be used to show the generality of new controllers or learned policies and ideally, be used by researchers to make their research more accessible and reproducible.

Cheers to the new year~! :clinking_glasses:

**Comment by yuvaltassa** [comment]
This is cool!

Can you make some kinematic videos from the CMU dataset, like [in this video](https://www.youtube.com/watch?v=rAai4QzcYbs&t=136s) (starting at 2:16)? 

The code you need is, I believe, [here](https://github.com/deepmind/dm_control/blob/main/dm_control/locomotion/mocap/cmu_mocap_data.py).

**Reply by rohanpsingh** [reply]
Do you mean perform a kinematic retargetting? That would be tough, isn't it? The robot (character) in the video is probably making motions beyond the joint ranges of JVRC1 and without ensuring dynamic stability, JVRC1 would simply fall. Maybe it would be possible to learn a policy for the same :slightly_smiling_face:

**Comment by yuvaltassa** [comment]
Another question: what are you using for visualisation? (specifically those collapsible interactive overlays)

**Reply by rohanpsingh** [reply]
The GUI menus are built using [Dear ImGui](https://github.com/ocornut/imgui/). The whole window you see is part of a software we created called "[mc-mujoco](https://github.com/rohanpsingh/mc_mujoco)" - which is an interface between [mc_rtc](https://jrl-umi3218.github.io/mc_rtc/index.html) (a robot framework for QP based controllers) and mujoco. Bipedal locomotion above is achieved using the [LIPM-Walking mc_rtc controller](https://github.com/stephane-caron/lipm_walking_controller).

**Reply by rohanpsingh** [reply]
One more example of using mc_mujoco is here: https://www.youtube.com/watch?v=7GmmAW5C20o
